### PR TITLE
Refactor box interpreter

### PR DESCRIPTION
**Pull request by emilmelnikov**
* Remove dead code
* Refactor BoxInterpreter.eventFilter and some of the surrounding code

### CODE REVIEW

File: ilastik/applets/counting/countingGuiBoxesInterface.py [hunk 1]
```diff
@@ -626,51 +623,36 @@ def release(self):
...
-
-        :param navigationInterpreter:
-        :param positionModel:
-        :param BoxContr:
-        :param widget: The main widget
-
-        """
-
-        QObject.__init__(self)
+    def __init__(self, base: NavigationInterpreter, posModel: PositionModel, mainWidget: QWidget):
```
Comment (k-dominik) [inline ilastik/applets/counting/countingGuiBoxesInterface.py]: not sure how `mainWidget` is more descriptive than just widget, but after consulting `QRubberBand` docs I'm not wiser ;). The Qt sourcecode gives me the feeling that it should be called `parent`

Comment (emilmelnikov) [reply ilastik/applets/counting/countingGuiBoxesInterface.py]: I'm afraid that calling this parameter `parent` would give the wrong impression that `BoxInterpreter` itself has a notion of a "parent" widget, something that `QObject` does.
How about `centralWidget`, just to be consistent with `LayerViewerGui`?

Comment (k-dominik) [reply ilastik/applets/counting/countingGuiBoxesInterface.py]: okay, but maybe add a docstring making clear that it's just the parent for the rubberwidget.

Comment (emilmelnikov) [reply ilastik/applets/counting/countingGuiBoxesInterface.py]: I ended up renaming it to `newBoxParent` with the corresponding docstring.

**Pull request merged**